mariamihaelafainarea/assignment1/SinglyLinkedList

```diff
@@ -0,0 +1,183 @@
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author maria
+ */
+//i cannot get rid of request changes thing. 
+public class SinglyLinkedList<T> {
+//i cannot get rid of request changes thing. 
+    //a class to test the problem !!!!
+    private SinglyLinkedListNode<T> first = null;
+    
+    public static class Books {
+        private String nume;
+
+        public Books(final String nume) {
+            this.nume = nume;
+        }
+
+        public Books() {
+        }
+        @Override
+        public String toString() {
+            return this.nume;
+        }
+
+    }
+    //English Explanation
+    //i need two pointers : first and last
+    //the first pointer is set on the first elem
+    //only when the last pointer came to the k + 1 -th position
+    //the first pointer will be incremented.
+    //In this way the first pointer
+    //will indicate at the end of the algorithm
+    //the kth element from the end to the beggining !
+
+    //i need a generic LinkedList
+    //Singly LinkedList
+    class SinglyLinkedListNode<T> {
+        private T value;
+        private SinglyLinkedListNode next;
+        public T getValue() {
+            return value;
+        }
+        public SinglyLinkedListNode getNext() {
+            return next;
+        }
+        public void setNext(SinglyLinkedListNode next) {
+            this.next = next;
+        }
+        public void setValue(T value) {
+            this.value = value;
+        } 
+        SinglyLinkedListNode(final T value, final SinglyLinkedListNode next) {
+            this.value = value;
+            this.next = next;
+        }
+    }
+    public SinglyLinkedListNode<T> getFirst() {
+        return first;
+    }
+    public void setFirst(SinglyLinkedListNode<T> first) {
+        this.first = first;
+    }
+    public SinglyLinkedList() {
+    }
+    final void insert(final T val) {
+        //insert an element to the end of the list
+        SinglyLinkedListNode<T> aux = first;
+        if (first == null) {
+            first = new SinglyLinkedListNode<>(val, null);
+        } else {
+            while (aux.next != null) {
+                aux = aux.next;
+            }
+            aux.next = new SinglyLinkedListNode<T>(val, null);
+            System.out.println("inserted");
+        }
+    }
+    final void addFirst(final T val) {
+        if (first == null) {
+            first = new SinglyLinkedListNode<>(val, null);
+        } else {
+            SinglyLinkedListNode<T> aux = new SinglyLinkedListNode<>(val, first);
+            first = aux;
+        }
+    }
+    final void addLast(final T val) {
+        insert(val);
+    }
+    final void remove(int index) {
+        //remove from a specific position
+        SinglyLinkedListNode<T> aux = first;
+        //first has the index == 0
+        if (first != null) {
+            while (index > 1 && aux != null) {
```

Comment:
Can you think what will happen for index == 0?

Reply:
Yes. For index zero it deleted in a sequence like "a b c " the "b" instead of a. I have modified it.